frontend/appflowy_flutter/lib/plugins/database_view/grid/presentation/grid_page.dart

```diff
@@ -216,67 +201,102 @@ class _GridRowsState extends State<_GridRows> {
 
   @override
   Widget build(BuildContext context) {
-    return BlocConsumer<GridBloc, GridState>(
-      listenWhen: (previous, current) => previous.reason != current.reason,
-      listener: (context, state) {
-        state.reason.whenOrNull(
-          insert: (item) {
-            _key.currentState?.insertItem(item.index);
-          },
-          delete: (item) {
-            _key.currentState?.removeItem(
-              item.index,
-              (context, animation) =>
-                  _renderRow(context, item.rowInfo, animation),
+    return BlocBuilder<GridFilterMenuBloc, GridFilterMenuState>(
+      builder: (context, filterState) {
+        return BlocBuilder<SortMenuBloc, SortMenuState>(
+          builder: (context, sortState) {
```

Comment:
an optional suggestion:

```dart
Builder(
  builder: ((context) {
    final filterState = context.watch<GridFilterMenuBloc>().state;
    final sortState = context.watch<SortMenuBloc>().state;
    return ...;
  })
)
```

this will reduce one `{}`.